rpcs3/Emu/RSX/RSXThread.cpp

```diff
@@ -295,24 +305,119 @@ namespace rsx
 						// Actually uses the modulo operator
 						_min_index = 0;
 						_max_index = attrib.frequency - 1;
+
+						if (freq_count != umax)
+						{
+							if (std::find(std::begin(frequencies), std::begin(frequencies) + freq_count, attrib.frequency) == std::begin(frequencies) + freq_count)
+							{
+								frequencies[freq_count++] = attrib.frequency;
```

Comment:
how is ensured that freq_count is not >= 16 ?

Reply:
This is a very hot path, if you can do without loops it would be best.